wallets/nwc/client.js

```diff
@@ -11,11 +11,8 @@ export async function testSendPayment ({ nwcUrl }, { logger }) {
 }
 
 export async function sendPayment (bolt11, { nwcUrl }, { logger }) {
-  const result = await nwcCall({
-    nwcUrl,
-    method: 'pay_invoice',
-    params: { invoice: bolt11 }
-  },
-  { logger })
+  const nwc = await getNwc(nwcUrl)
+  const { error, result } = await nwc.payInvoice(bolt11)
+  if (error) throw new Error(error.code + ' ' + error.message)
```

Comment:
what's the point of catching the exception here? If it throw inside payInvoice it will be just propagated back because we are awaiting it.
The  `if (error) throw new Error(error.code + ' ' + error.message)` makes sure that if ndk decides to start following their own type specs,  we won't leave any error unthrown.

Reply:
> what's the point of catching the exception here? If it throws inside payInvoice it will be just propagated back because we are awaiting it.

It throws an object instead of a proper error message so we log `payment failed: [Object object]` currently

> The if (error) throw new Error(error.code + ' ' + error.message) makes sure that if ndk decides to start following their own type specs, we won't leave any error unthrown.

okay, we can keep this then but we still need `try/catch` currently

Reply:
Ah! Ok, that's a pretty strange way to pass errors... i'm adding the recatch

Reply:
done